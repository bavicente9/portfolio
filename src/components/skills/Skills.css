.skills {
    padding: 50px 0;
    background: var(--LigthGray);
}
.skills > h1{
    font-size: 2rem;
    text-transform: uppercase;
    text-align: center;
    color: var(--DarkOrange);
    margin-bottom: 50px;
}

.skills__container {
    padding: 5px var(--GeneralMarginWidth);
    display: grid;
    align-items: center;
    justify-content: center;

    grid-template-columns: repeat(auto-fit, 100px);
    grid-template-rows: repeat(auto-fit, 175px);
    
    list-style: none;
    gap: 25px;
}

.skills__item{
    height: 175px;
    width: 100px;
    background: var(--GeneralBackGround);
    
    display: grid;
    grid-template-columns: 100px;
    grid-template-rows: 100px 75px;
    
    border: 1px solid var(--Orange);
    border-radius: 15px;
    overflow: hidden;
    transition-duration: 0.34s;
    
    color: #fff;
}


.skills__item:hover{
    transform: scale(1.2);
}

.skills__itemImg{
    background: #fff;
    height: 100px;
    max-height: 100px;
    width: 100px;
}
.skills__itemImg img{
    height: 100%;
    width: 100%;
    object-fit:scale-down;
}

.skills__itemName{
    margin: 0;
    place-self: center;
    font-family: 'Roboto', sans-serif;
    font-size: 1rem;
    text-transform: uppercase;
}

@media (max-width:720px){
    .skills__container{
        padding: 5px var(--GeneralRMarginWidth);
    }
}