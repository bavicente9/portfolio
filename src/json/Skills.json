[
    {
        "name": "React",
        "img": "./images/icons/icon_react1.svg"
    },
    {
        "name": "JavaScript",
        "img": "./images/icons/icon_javaScript.svg"
    },
    {
        "name": "CSS",
        "img": "./images/icons/icon_css.svg"
    },
    {
        "name": "HTML5",
        "img": "./images/icons/icon_html.svg"
    },
    {
        "name": "Java",
        "img": "./images/icons/icon_java.svg"
    }
]