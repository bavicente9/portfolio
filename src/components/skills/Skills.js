import './Skills.css'


export const Skills = ({ infoJson }) => {

    return (
        <section aria-label='skills' id='skills' className='skills'>
            <h1>Technologies</h1>
            <ul className='skills__container' aria-label='list of skills'>
                {
                    //generate the skills
                    infoJson.map((item, index) => {
                        return (
                            <li key={index} className='skills__item'>
                                <span className='skills__itemImg'><img alt='' src={item.img} /></span>
                                <p className='skills__itemName'>{item.name}</p>
                            </li>
                        )
                    })
                }
            </ul>
        </section>
    )
}
