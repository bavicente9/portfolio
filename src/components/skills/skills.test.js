import { Skills } from "./Skills"
import { render } from '@testing-library/react'

describe('Skills', () => {
    const infoJson = JSON.parse(`
    [{
        "name": "React",
        "img": "./images/icons/icon_react1.svg"
    },
    {
        "name": "JavaScript",
        "img": "./images/icons/icon_javaScript.svg"
    },
    {
        "name": "CSS",
        "img": "./images/icons/icon_css.svg"
    }]`
    )

    it('renders correctly', () => {
        const { asFragment } = render(<Skills infoJson={infoJson} />)
        expect(asFragment()).toMatchSnapshot()
    })
})