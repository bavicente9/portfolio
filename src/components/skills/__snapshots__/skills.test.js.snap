// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills renders correctly 1`] = `
<DocumentFragment>
  <section
    aria-label="skills"
    class="skills"
    id="skills"
  >
    <h1>
      Skills
    </h1>
    <ul
      aria-label="list of skills"
      class="skills__container"
    >
      <li
        class="skills__item"
      >
        <span
          class="skills__itemImg"
        >
          <img
            alt=""
            src="./images/icons/icon_react1.svg"
          />
        </span>
        <p
          class="skills__itemName"
        >
          React
        </p>
      </li>
      <li
        class="skills__item"
      >
        <span
          class="skills__itemImg"
        >
          <img
            alt=""
            src="./images/icons/icon_javaScript.svg"
          />
        </span>
        <p
          class="skills__itemName"
        >
          JavaScript
        </p>
      </li>
      <li
        class="skills__item"
      >
        <span
          class="skills__itemImg"
        >
          <img
            alt=""
            src="./images/icons/icon_css.svg"
          />
        </span>
        <p
          class="skills__itemName"
        >
          CSS
        </p>
      </li>
    </ul>
  </section>
</DocumentFragment>
`;
